test/common/http/header_mutation_test.cc

```diff
@@ -0,0 +1,297 @@
+#include "source/common/http/header_mutation.h"
```

Comment:
Do you have any integration tests, in particular ones that cover the upstream filter use case?

Reply:
I will add some integration tests for this filter.